public/pages/workflow_detail/components/header.tsx

```diff
@@ -52,14 +73,130 @@ export function WorkflowDetailHeader(props: WorkflowDetailHeaderProps) {
     }
   }, [props.workflow]);
 
-  return (
+  const { TopNavMenu, HeaderControl } = getNavigationUI();
+  const { setAppRightControls } = getApplication();
+  const {
+    chrome: { setHeaderVariant },
+  } = getCore();
+  const uiSettings = getUISettings();
+  const showActionsInHeader = uiSettings.get('home:useNewHomePage');
+
+  useEffect(() => {
+    if (showActionsInHeader) {
+      setHeaderVariant?.(HeaderVariant.APPLICATION);
+    }
+
+    return () => {
+      setHeaderVariant?.();
+    };
+  }, [setHeaderVariant, showActionsInHeader]);
+
+  const onExportButtonClick = () => {
+    setIsExportModalOpen(true);
+  };
+
+  const onExitButtonClick = () => {
+    history.replace(
+      `${APP_PATH.WORKFLOWS}${
+        dataSourceId !== undefined ? `?dataSourceId=${dataSourceId}` : ''
+      }`
+    );
+  };
+
+  const topNavConfig: TopNavMenuData[] = [
+    {
+      iconType: 'exportAction',
+      tooltip: 'Export',
+      ariaLabel: 'Export',
+      run: onExportButtonClick,
+      controlType: 'icon',
+    } as TopNavMenuIconData,
+    {
+      iconType: 'exit',
+      tooltip: 'Return to projects',
+      ariaLabel: 'Exit',
+      run: onExitButtonClick,
+      controlType: 'icon',
+    } as TopNavMenuIconData,
+  ];
+
+  const dataSourceEnabled = getDataSourceEnabled().enabled;
+  const dataSourceId = getDataSourceId();
```

Comment:
For consistency, I would suggest moving these types of vars at the top of the fn, like around line 59. Same with the variables on lines 76-82. Then we can have all of those state vars etc in one place. Additionally, we want them defined before custom `useEffect` hooks typically, since the execution order matters. If these are defined/updated later, could cause weird state bugs.